youtube_dl/extractor/seeso.py

```diff
@@ -0,0 +1,91 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+import urlparse
+import os.path
+import json
+from ..utils import (
+    sanitized_Request,
+    urlencode_postdata
+)
+
+
+class SeesoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?seeso\.com/view/episode/(?P<id>[0-9]+)'
+    _TEST = {
+        'params': {
+            'username': 'emailhere',
+            'password': 'passwordhere'
+        },
+        'url': 'https://www.seeso.com/view/episode/799241283849',
+        'info_dict': {
+            'id': '799241283849',
+            'ext': 'mp4',
+            'series': 'Bajillion Dollar Propertie$',
+            'title': 'Farsi Lessons',
+            'description': ("Amir leads Victoria into a trap, the Bros meet a super obnoxious bro, "
+                            "and rival brokers Serge and Gio intimidate Glenn."),
+            'thumbnail': 'https://chaosic.akamaized.net/NBCOTT_-_Production/360/907/160831_3092487_Farsi_Lessons.jpg',
+        }
+    }
+
+    def _real_extract(self, url):
+        username, password = self._get_login_info()
+        video_id = self._match_id(url)
+
+        if username is None or password is None:
+            return
+
+        _API_AUTH_URL = "https://www.seeso.com/api/v3/auth/login"
+        auth_json = {
+            "username": username,
+            "password": password,
+            "userOnly": True,
+            "isRememberMe": True,
+            "isWeb": True
+        }
+
+        # Send auth POST request and get token from response
+        auth_request = sanitized_Request(_API_AUTH_URL, urlencode_postdata(auth_json))
+        auth_response = json.loads(self._download_webpage(auth_request, '', note='Getting auth token...'))
+        auth_token = auth_response.get('user').get('token')
+
+        # Use the public unauthenticated API to get the video's info
+        _VIDEO_INFO_URL = 'https://feed.theplatform.com/f/NZILfC/nbcott-prod-all-media?byAvailabilityState=' \
+                          'available&byId=%s&form=cjson'
+        request = sanitized_Request(_VIDEO_INFO_URL % video_id)
+        json_data = self._download_json(request, video_id)
+        entry = json_data.get('entries')[0]
+
+        # Template fields
+        series = entry["nbc-chaos$show"]                    # Show name
+        title = entry["nbc-chaos$shortTitle"]               # Episode Name
+        season_number = entry["nbc-chaos$seasonNumber"]
+        episode_number = entry["nbc-chaos$episodeNumber"]
+        thumbnail = entry["defaultThumbnailUrl"]
+        description = entry["nbc-chaos$shortDescription"]
```

Comment:
None of these fields apart from `title` is allowed to break extraction when missing.